catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/operations/AirflowPipelineResource.java

```diff
@@ -393,7 +393,7 @@ public Response createOrUpdate(
     AirflowPipeline pipeline = getAirflowPipeline(securityContext, update);
     SecurityUtil.checkAdminRoleOrPermissions(authorizer, securityContext, dao.getOriginalOwner(pipeline));
     PutResponse<AirflowPipeline> response = dao.createOrUpdate(uriInfo, pipeline);
-    deploy(pipeline);
+    deploy(pipeline, SecurityUtil.isAdminOrBotRole(authorizer, securityContext));
```

Comment:
@harshach 
Decrypt only if admin or bot, not it is a normal user owning the DatabaseService.

Reply:
I don't think we ever allow non admin/bot to create a service. For now lets only decrypt if the user is admin or bot

Reply:
Ok